src/main/java/frc/robot/elevator/Elevator.java

```diff
@@ -174,19 +215,39 @@ public void fastPeriodic() {
           RobotConstants.FAST_PERIODIC_DURATION, 
           profiledSetpoint, 
           new TrapezoidProfile.State(targetState.getPosMeters(), 0));
-
-      positionController.setSetpoint(
-          Rotations.of(profiledSetpoint.position / rotationsToMeters), 
-          RotationsPerSecond.of(profiledSetpoint.velocity / rotationsToMeters));
-    } else if (currentState != ElevatorState.STOW) {
-      // If reached target state, and that state isn't stowed (we have special behavior for that), set setpont to hold at that state
+    } else {
+      // If reached target state, set setpont to hold at that state
       profiledSetpoint = new TrapezoidProfile.State(targetState.getPosMeters(), 0.0);
-      positionController.setSetpoint(
-          Rotations.of(targetState.getPosMeters() / rotationsToMeters), 
-          RotationsPerSecond.of(0));
     }
 
-    positionController.periodic();
+    if (voltageSet == false) {
+      // If no manual voltage set, calculate voltage using feedforward and feedback
+      double feedforwardVolts = feedforwardController.calculate(profiledSetpoint.velocity);
+      double feedbackVolts = 0.0;
+      
+      if (MathUtil.isNear(0.0, motorValues.velocity.in(RotationsPerSecond) * rotationsToMeters, PIDThreshold.in(MetersPerSecond))) {
+        // If target velocity is close enough to zero, meaning you are reacing the end of a trajectory, fade in some feedback voltage
+        feedbackVolts = feedbackController.calculate(motorValues.position.in(Rotations) * rotationsToMeters, profiledSetpoint.position);
+        feedbackVolts *= Math.pow((Math.abs(motorValues.velocity.in(RotationsPerSecond) * rotationsToMeters)/PIDThreshold.in(MetersPerSecond)) * -1 + 1, PIDSmoothing); // This is what does the fading in
+      }
+
+      voltageOut.mut_replace(feedforwardVolts + feedbackVolts, Volts);
+    } else {
+      // Otherwise, dont change voltageOut at all, and reset voltageSet
+      voltageSet = false;
```

Comment:
This will set voltageSet to false if voltageSet was true. Is this the intended effect?